app/sidekiq/event_bus_gateway/letter_ready_email_job.rb

```diff
@@ -16,35 +15,63 @@ class LetterReadyEmailJob
     }.freeze
 
     def perform(participant_id, template_id)
-      notify_client.send_email(
-        recipient_identifier: { id_value: participant_id, id_type: 'PID' },
-        template_id:,
-        personalisation: { host: HOSTNAME_MAPPING[Settings.hostname] || Settings.hostname,
-                           first_name: get_first_name_from_participant_id(participant_id) }
-      )
+      if get_mpi_profile(participant_id)
+        response = notify_client.send_email(
+          recipient_identifier: { id_value: participant_id, id_type: 'PID' },
+          template_id:,
+          personalisation: { host: HOSTNAME_MAPPING[Settings.hostname] || Settings.hostname,
+                             first_name: get_first_name_from_participant_id(participant_id) }
+        )
+        EventBusGatewayNotification.create(user_account: user_account(participant_id), template_id:,
+                                           va_notify_id: response.id)
+      end
     rescue => e
       record_email_send_failure(e)
     end
 
     private
 
     def notify_client
-      VaNotify::Service.new(NOTIFY_SETTINGS.api_key,
-                            { callback_klass: 'EventBusGateway::VANotifyEmailStatusCallback' })
+      @notify_client ||= VaNotify::Service.new(NOTIFY_SETTINGS.api_key,
+                                               { callback_klass: 'EventBusGateway::VANotifyEmailStatusCallback' })
     end
 
-    def get_first_name_from_participant_id(participant_id)
-      bgs = BGS::Services.new(external_uid: participant_id, external_key: participant_id)
-      person = bgs.people.find_person_by_ptcpnt_id(participant_id)
-      if person
-        person[:first_nm].capitalize
-      else
-        raise StandardError, 'Participant ID cannot be found in BGS'
+    def user_account(participant_id)
```

Comment:
Fundamentally, this is the part of the code that is responsible for the subsequent MPI lookup. The logic goes like this:

I am assuming we cannot save the veteran participant_id directly on the EventBusGatewayNotification. As such,
We can save an association to a UserAccount, which has an ICN but not a participant ID, and since
Event Bus feeds us a participant_id, we have to use that participant ID to look up the ICN, so
We route it through MPI